edb/lang/schema/scalars.py

```diff
@@ -228,13 +228,21 @@ def _is_assignment_castable_impl(source: str, target: str) -> bool:
     ],
 
     ast.ops.DIV: [
-        ('std::int64', 'std::int64', 'std::int64'),
+        ('std::int64', 'std::int64', 'std::float64'),
         ('std::float32', 'std::float32', 'std::float32'),
         ('std::float64', 'std::float64', 'std::float64'),
         ('std::decimal', 'std::decimal', 'std::decimal'),
         ('std::decimal', 'std::int64', 'std::decimal'),
     ],
 
+    ast.ops.FLOORDIV: [
+        ('std::int64', 'std::int64', 'std::int64'),
+        ('std::float32', 'std::float32', 'std::int64'),
```

Comment:
We want the result of `<float32>1 // <float32>2 + <float32>1` to be a `float32`.